x-pack/plugins/reporting/public/lib/reporting_api_client/reporting_api_client.ts

```diff
@@ -71,68 +79,75 @@ export class ReportingAPIClient {
     });
   }
 
-  public list = (page = 0, jobIds: string[] = []): Promise<JobQueueEntry[]> => {
+  public async list(page = 0, jobIds: string[] = []) {
     const query = { page } as any;
     if (jobIds.length > 0) {
       // Only getting the first 10, to prevent URL overflows
       query.ids = jobIds.slice(0, 10).join(',');
     }
 
-    return this.http.get(`${API_LIST_URL}/list`, {
+    const jobQueueEntries: ReportApiJSON[] = await this.http.get(`${API_LIST_URL}/list`, {
       query,
       asSystemRequest: true,
     });
-  };
 
-  public total(): Promise<number> {
-    return this.http.get(`${API_LIST_URL}/count`, {
+    return jobQueueEntries.map((report) => new Job(report));
+  }
+
+  public async total() {
+    return await this.http.get(`${API_LIST_URL}/count`, {
       asSystemRequest: true,
     });
   }
 
-  public getContent(jobId: string): Promise<JobContent> {
-    return this.http.get(`${API_LIST_URL}/output/${jobId}`, {
+  public async getError(jobId: string) {
+    return await this.http.get(`${API_LIST_URL}/output/${jobId}`, {
       asSystemRequest: true,
     });
   }
 
-  public getInfo(jobId: string): Promise<ReportApiJSON> {
-    return this.http.get(`${API_LIST_URL}/info/${jobId}`, {
+  public async getInfo(jobId: string) {
+    const report: ReportApiJSON = await this.http.get(`${API_LIST_URL}/info/${jobId}`, {
       asSystemRequest: true,
     });
+    return new Job(report);
   }
 
-  public findForJobIds = (jobIds: JobId[]): Promise<ReportDocument[]> => {
-    return this.http.fetch(`${API_LIST_URL}/list`, {
+  public async findForJobIds(jobIds: JobId[]) {
+    const reports: ReportApiJSON[] = await this.http.fetch(`${API_LIST_URL}/list`, {
       query: { page: 0, ids: jobIds.join(',') },
       method: 'GET',
     });
-  };
+    return reports.map((report) => new Job(report));
+  }
 
   /*
    * Return a URL to queue a job, with the job params encoded in the query string of the URL. Used for copying POST URL
    */
```

Comment:
We should move these documentation strings to the `IReportingAPI` interface above. WDYT?

Reply:
5cf4e221553